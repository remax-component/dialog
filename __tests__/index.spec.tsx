import "expect-puppeteer";
import React from "react";
import Dialog from "../src/";
import { render } from "@remax/runtime";
import Container from "@remax/runtime/cjs/Container";
import { View } from "remax/one";

const p = {
  setData(state: any, callback: Function) {
    setTimeout(() => {
      if (typeof callback === "function") {
        callback();
      }
    });
  },
  $batchedUpdates(callback: Function) {
    callback();
  },
  $spliceData(state: any, callback: Function) {
    setTimeout(() => {
      if (typeof callback === "function") {
        callback();
      }
    });
  },
};

describe("Dialog", () => {
  class SimpleDialog extends React.Component {
    render() {
      return <Dialog title="233" visible />;
    }
  }

  it("can render", () => {
    const container = new Container(p);
    render(<SimpleDialog />, container);
    expect(container.root).toMatchSnapshot();
  });
});
