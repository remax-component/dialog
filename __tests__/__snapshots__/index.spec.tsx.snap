// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog can render 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "id": 1,
          "props": Object {
            "class": "rmc-dialog-mask",
            "style": "",
          },
          "text": undefined,
          "type": "view",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "text": "233",
                              "type": "plain-text",
                            },
                          ],
                          "id": 3,
                          "props": Object {
                            "class": "rmc-dialog-title",
                          },
                          "text": undefined,
                          "type": "view",
                        },
                      ],
                      "id": 4,
                      "props": Object {
                        "class": "rmc-dialog-header",
                      },
                      "text": undefined,
                      "type": "view",
                    },
                    Object {
                      "children": Array [],
                      "id": 5,
                      "props": Object {
                        "class": "rmc-dialog-body",
                        "style": "",
                      },
                      "text": undefined,
                      "type": "view",
                    },
                  ],
                  "id": 6,
                  "props": Object {
                    "class": "rmc-dialog-content",
                  },
                  "text": undefined,
                  "type": "view",
                },
              ],
              "id": 7,
              "props": Object {
                "class": "rmc-dialog",
                "role": "document",
              },
              "text": undefined,
              "type": "view",
            },
          ],
          "id": 8,
          "props": Object {
            "class": "rmc-dialog-wrap",
            "data-wrapper": true,
            "onTap": "$$REMAX_METHOD_8_onTap",
            "role": "dialog",
            "style": "",
          },
          "text": undefined,
          "type": "view",
        },
      ],
      "id": 9,
      "props": Object {},
      "text": undefined,
      "type": "view",
    },
  ],
  "id": 0,
  "props": undefined,
  "text": undefined,
  "type": "root",
}
`;
